### PR TITLE
ci: Remove android, ios, ROCm, arm64, cu101 builds

### PR DESCRIPTION
**Pull request by kere-nel**
removing LTS unsupported builds -  macos, libtorch, cpu, rocm, cu101

### CODE REVIEW

**Comment by antoniovs1029** [comment]
Seems your branch doesn't include the changes I introduced on https://github.com/pytorch/pytorch/pull/57362 In particular, your builds aren't checking out the 1.8 version of `pytorch/builder`, so my guess is that the failing builds are related to that.

**Review by malfet — CHANGES_REQUESTED** [review]
Can you please elaborate the reason behind each removal?
I.e. cu101 (not supported) and rocm (beta) sounds reasonable.
But Python-3.6 and MacOS needs some discussion.
Same about ASAN config: what is the motivation behind it?

**Comment by Lynx1820** [comment]
> Can you please elaborate the reason behind each removal?
> I.e. cu101 (not supported) and rocm (beta) sounds reasonable.
> But Python-3.6 and MacOS needs some discussion.
> Same about ASAN config: what is the motivation behind it?

We were only planning to test cuda builds (10.2 and 11.1), so that's why I removed ASAN configs, since they don't seem to be run on cuda. However, I have added them back. @harishsk, does these subset of builds look ok, or should I remove any others?

File: .circleci/cimodel/data/simple/ge_config_tests.py [hunk 1]
```diff
@@ -57,20 +56,6 @@ def gen_tree(self):
...
-        ["jit_legacy", "test"],
-        ["pytorch_linux_xenial_py3_6_gcc5_4_build"]),
-    GeConfigTestJob(
-        None,
-        None,
-        CudaVersion(10, 2),
-        ["cudnn7", "py3", "jit_legacy", "test"],
-        ["pytorch_linux_xenial_cuda10_2_cudnn7_py3_gcc7_build"],
-        use_cuda_docker=True,
-    ),
```
Comment (malfet) [inline .circleci/cimodel/data/simple/ge_config_tests.py:73]: Why this configuration is getting removed? (which tests some of the jit legacy modes?)

**Review by malfet — APPROVED** [review]

**Comment by Lynx1820** [comment]
@malfet do you know if it's possible to disable the `pr/pytorch-linux-bionic-rocm4.1-py3.6` build?

**Comment by seemethere** [comment]
> @malfet do you know if it's possible to disable the `pr/pytorch-linux-bionic-rocm4.1-py3.6` build?

@jithunnair-amd or @jeffdaily is it possible to disable this jenkins job for particular branches?

**Comment by jeffdaily** [comment]
> > @malfet do you know if it's possible to disable the `pr/pytorch-linux-bionic-rocm4.1-py3.6` build?
> 
> @jithunnair-amd or @jeffdaily is it possible to disable this jenkins job for particular branches?

Why do we want to disable our Jenkins job for PRs?

**Comment by Lynx1820** [comment]
> > > @malfet do you know if it's possible to disable the `pr/pytorch-linux-bionic-rocm4.1-py3.6` build?
> > 
> > 
> > @jithunnair-amd or @jeffdaily is it possible to disable this jenkins job for particular branches?
> 
> Why do we want to disable our Jenkins job for PRs?

The LTS branch will not be responsible for supporting rocm, so we're trying to remove all builds that include this configuration, which includes the Jenkins job.

**Comment by Lynx1820** [comment]
> > > @malfet do you know if it's possible to disable the `pr/pytorch-linux-bionic-rocm4.1-py3.6` build?
> > 
> > 
> > @jithunnair-amd or @jeffdaily is it possible to disable this jenkins job for particular branches?
> 
> Why do we want to disable our Jenkins job for PRs?

@jeffdaily any updates on this?

**Comment by jeffdaily** [comment]
@Lynx1820 , ossci-job-dsl that controls our Jenkins jobs was updated by @seemethere in https://github.com/pytorch/ossci-job-dsl/pull/104.

**Review by malfet — APPROVED** [review]

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**